src/Compilers/Core/Portable/SourceGeneration/ISourceGenerator.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+#nullable enable
+namespace Microsoft.CodeAnalysis
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "In progress")]
```

Comment:
>System.Diagnostics.CodeAnalysis.SuppressMessage [](start = 5, length = 47)

Consider adding PROTOTYPE comment to indicate temporary nature of each suppression like this.

Reply:
We have an issue on the board tracking the overall shape for public api and removing suppressions.

---
In reply to: [382807788](https://github.com/dotnet/roslyn/pull/41789#discussion_r382807788) [](ancestors = 382807788)